backend/src/acidwatch_api/app.py

```diff
@@ -128,11 +126,19 @@ async def run_model(
     model_id: str,
     request: RunRequest,
     jwt_token: Annotated[str | None, Security(get_jwt_token)],
-) -> SimulationResults:
+) -> RunResponse:
     adapter_class = ADAPTERS[model_id]
     adapter = adapter_class(request.concs, request.settings, jwt_token)
-    result = await adapter.run()
-    return _legacy.result_to_simulation_results(request.concs, result)
+    runid = uuid4()
+    await _run_adapter(adapter, runid)
+    result = RESULTS[runid]
+
+    if isinstance(result, BaseException):
+        raise HTTPException(
+            status_code=400,
+            detail=f"Model failed to calculate the change: {format_exception(result)}",
+        )
```

Comment:
I made the status code 400, I think it's probably incorrect.. It might not be bad request just because the model fails. I wondered about 500 as well, but it's not really internal error, is it?

Reply:
A bit further down we do:

``` 
   if isinstance(result, BaseException):
        return {"error": format_exception(result)}
```

We probably want to be consistent in how we handle errors in our api. We could change to this or do fixes afterwards - we don't need to do everything now.